app/views/characters/_group.haml

```diff
@@ -8,8 +8,8 @@
       = group ? "Group: #{group.name}" : 'Ungrouped Characters'
 - if characters.try(:exists?)
   - template_ids = characters.group(:template_id).pluck(:template_id).uniq.compact
-  - ::Template.where(id: template_ids).ordered.each do |template|
-    - next if skip_grouped_templates && template.characters.where.not(character_group_id: nil).exists?
+  - ::Template.select('*', 'exists (select * from characters where characters.template_id = templates.id and characters.character_group_id is not NULL) as has_non_null_character').where(id: template_ids).ordered.each do |template|
+    - next if skip_grouped_templates && template.has_non_null_character
```

Comment:
Hm, this feels like there should be a less... obscure... way to check this. Also we hardly ever load character group view but that's not an excuse.

Reply:
My Rails SQL skills are not expert; I would be completely unsurprised to learn that there's a more elegant way to do this.

I've definitely seen other `selects` in the code though. Do you have a suggestion of what would be better?

Reply:
I was just looking into this! it looks like possibly rails expects you to do something like
```suggestion
  - templates = ::Template.where(id: template_ids)
  - templates.where.not(id: users.characters.where('character_group_id IS NOT NULL').select(:template_id).uniq) if skip_grouped_templates
  - templates.ordered.each do |template|
```
or similar. since they did stuff to their exists method. 

`.where.not(character_group_id: nil` *ought* to work but iirc instead does something weird.

Anyway i think that's approximately the same (selects templates that have characters with a group id and excludes them) and postgres is smart enough to do it with a join if that's more efficent.